src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -122,14 +123,32 @@ private void Commit(CompletionItem item, TextChange textChange, Model model, cha
 
             var document = this.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
             var formattingService = document.GetLanguageService<IEditorFormattingService>();
-            if (formattingService != null &&
-                (item.ShouldFormatOnCommit || (commitChar != null && formattingService.SupportsFormattingOnTypedCharacter(document, commitChar.GetValueOrDefault()))))
+
+            var commitCharTriggersFormatting = commitChar != null && 
+                    (formattingService?.SupportsFormattingOnTypedCharacter(document, commitChar.GetValueOrDefault()) 
+                     ?? false);
+
+            if (formattingService != null && item.ShouldFormatOnCommit || commitCharTriggersFormatting)
```

Comment:
please parenthesize this appropriately to make it clearer.  (i hate mixed &&'s and ||'s in the same expression).